conjure_oxide/tests/generated_tests.rs

```diff
@@ -229,37 +229,38 @@ fn integration_test_inner(
 }
 
 fn assert_vector_operators_have_partially_evaluated(model: &conjure_core::Model) {
-    model.constraints.transform(Arc::new(|x| {
-        use conjure_core::ast::Expression::*;
-        match &x {
-            Bubble(_, _, _) => (),
-            FactorE(_, _) => (),
-            Sum(_, vec) => assert_constants_leq_one(&x, vec),
-            Min(_, vec) => assert_constants_leq_one(&x, vec),
-            Max(_, vec) => assert_constants_leq_one(&x, vec),
-            Not(_, _) => (),
-            Or(_, vec) => assert_constants_leq_one(&x, vec),
-            And(_, vec) => assert_constants_leq_one(&x, vec),
-            Eq(_, _, _) => (),
-            Neq(_, _, _) => (),
-            Geq(_, _, _) => (),
-            Leq(_, _, _) => (),
-            Gt(_, _, _) => (),
-            Lt(_, _, _) => (),
-            SafeDiv(_, _, _) => (),
-            UnsafeDiv(_, _, _) => (),
-            SumEq(_, vec, _) => assert_constants_leq_one(&x, vec),
-            SumGeq(_, vec, _) => assert_constants_leq_one(&x, vec),
-            SumLeq(_, vec, _) => assert_constants_leq_one(&x, vec),
-            DivEq(_, _, _, _) => (),
-            Ineq(_, _, _, _) => (),
-            // this is a vector operation, but we don't want to fold values into each-other in this
-            // one
-            AllDiff(_, _) => (),
-            WatchedLiteral(_, _, _) => (),
-            Reify(_, _, _) => (),
-        };
-        x.clone()
+    // Transform all constraints in the model
+    model.constraints.transform(Arc::new(|expressions| {
```

Comment:
due to uniplate this should work without an explicit for loop, i.e. the previous version unchanged. does it?

when I say unchanged, I think you would need to change transform to transform_bi

also, since this is an assertion and it is not going to update the model, we should probably change it to use universe/universe_bi.

Reply:
Hey! Sorry for not responding to your comment. I didn't see it. I agree with your comment. Should I start using universe_bi from now on, or its better to do it in a separate PR so nobody gets confused by this change?

Reply:
```rs
for <Vec<Expr> as Biplate<Expr>>::universe_bi(model.constraints) {
 ...
}
```

Reply:
we should implement biplate<expr> for model at some point, but that will be best done once we get struct derive working.

Reply:
If _ above doesnt work (I've never tried it), explicitly write Vec<Expr> in its place

Reply:
@ozgurakgun Should I open a separte PR on this?

Reply:
@ozgurakgun Should I do this in a separate PR, or this one?